test/lib/rules/no-unguarded-async-and-event-operations.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { testRule, testTypeScript } = require('../shared');
+
+testRule('no-unguarded-async-and-event-operations', {
+    valid: [
+        {
+            code: `if (!import.meta.env.SSR) { fetch('/api/data'); }`,
+        },
+        {
+            code: `async function fetchData() { if (!import.meta.env.SSR) { await fetch('/api/data'); } }`,
+        },
+        {
+            code: `if (!import.meta.env.SSR) { window.addEventListener('click', () => console.log('Clicked')); }`,
+        },
+        {
+            code: `if (!import.meta.env.SSR) { const response = fetch('/api/data'); }`,
+        },
+        {
+            code: `setTimeout(() => { console.log('Hello'); }, 1000);`, // This should still be valid
+        },
+        {
+            code: `if (!import.meta.env.SSR) { new Promise((resolve) => resolve('test')); }`,
+        },
+        {
+            code: `if (!import.meta.env.SSR) { somePromise.then(() => {}).catch(() => {}); }`,
+        },
+        {
+            code: `if (!import.meta.env.SSR) { this.dispatchEvent(new CustomEvent('myevent')); }`,
```

Comment:
The README uses `if (typeof window !== 'undefined')`. Is there a test for that style guard?

Reply:
Thats a miss from my side will update that